### DISCUSSION TITLE
Need help about Library

### DISCUSSION THREAD

**Discussion by Repulsar19** [JAndrassy/ArduinoOTA#211] [General]
Hello Sir, sorry I m new here ,so don't knew about how to send message to you. I want to flash arduino program to my STM32 Bluepill over Serial communication via ESP8266 Wifi ( without webserver uploading and without internet as Soft Access Point Mode). I cant able to understand how to achieve this . Can you please help me in this sirr ? I have also tried your Library "ArduinoOTA" but some points I don't understand .

**Comment by JAndrassy** [comment]
first you have to make the WiFiEspAT library work on your Bluepill
https://github.com/JAndrassy/WiFiEspAT#getting-started

**Reply by Repulsar19** [reply]
ok gonna check it down Thank You sir :)

**Reply by Repulsar19** [reply]
Hello Sir, sorry again to ask I can't able to understand about checking with STM32. Like first do I need to add new firmware for ESP8266 or do I need to upload code on STM32. I'm noob in this field and just entered into Embedded for studying it. Can you please guide me step by step please sir?
